Document/0x05b-Basic-Security_Testing.md

```diff
@@ -345,6 +345,76 @@ Successfully installed 1 modules, 0 already installed
 This will install any module that matches your query. Newly installed modules are dynamically loaded into the
 console and are available for immediate use.
 
+#### Firebase/Google Cloud Messaging (FCM/GCM)
+
+Firebase Cloud Messaging (FCM) is the successor of Google Cloud Messaging (GCM) and is a free service offered by Google and allows to send messages between an application server and client apps. The server and client app are communicating via the FCM/GCM connection server that is handling the downstream and upstream messages.
+
+![Architectural Overview](Images/Chapters/0x05b/FCM-notifications-overview.png)
+
+Downstream messages are sent from the application server to the client app (Push notifications); Upstream messages are sent from the client app to the server.
+
+FCM is available for Android and also for iOS and Chrome and can therefore be used on different platforms. FCM provides two connection server protocols at the moment: HTTP and XMPP. There are several differences in the implementation of HTTP And XMPP<sup>[24]</sup> when using them in FCM. The following example focuses on intercepting FCM using HTTP.
+
+##### Preparation
+
+For a full dynamic analysis of an Android app also FCM should be intercepted. To be able to intercept the messages several steps should be considered for preparation. The following example can be used on Mac OS X:
+
+* Install the CA certificate of your interception proxy into your Android phone<sup>[2]</sup>.
+* A Man-in-the-middle attack should be executed so all traffic from the mobile device is redirected to your testing machine. This can be done by using a tool like ettercap<sup>[24]</sup>. It can be installed by using brew.
```

Comment:
I think these two steps are not Mac OS specific, up to "It can be installed by using brew.". Maybe you can make these two steps general, and then continue here with something like "On Mac OS, you can use brew to install ettercap".